src/Core/Util/Form.php

```diff
@@ -0,0 +1,444 @@
+<?php
+/**
+ * BEdita, API-first content management framework
+ * Copyright 2018 ChannelWeb Srl, Chialab Srl
+ *
+ * This file is part of BEdita: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * See LICENSE.LGPL or <http://gnu.org/licenses/lgpl-3.0.html> for more details.
+ */
+
+namespace App\Core\Util;
+
+use Cake\Core\Configure;
+use Cake\Utility\Hash;
+use Cake\Utility\Inflector;
+
+class Form
+{
+    /**
+     * Schema property types
+     */
+    public const SCHEMA_PROPERTY_TYPES = ['string', 'number', 'integer', 'boolean', 'array', 'object'];
+
+    /**
+     * Custom controls
+     */
+    public const CUSTOM_CONTROLS = ['lang', 'status', 'old_password', 'password', 'confirm-password', 'title'];
+
+    /**
+     * Control types
+     */
+    public const CONTROL_TYPES = ['json', 'textarea', 'date-time', 'date', 'checkbox', 'enum'];
+
+    /**
+     * Get control by schema, control type, and value
+     *
+     * @param array $schema Object schema array.
+     * @param string $type Control type.
+     * @param mixed|null $value Property value.
+     * @return array
+     */
+    public static function control(array $schema, string $type, $value) : array
+    {
+        if (!in_array($type, Form::CONTROL_TYPES)) {
+            return compact('type', 'value');
+        }
+        $method = sprintf('%sControl', $type);
+        $method = str_replace('-', '', $method);
```

Comment:
What about 
```php
$method = sprintf('%sControl', Inflector::variable(str_replace('-', '_', $method)));
```

it should transform `my_control-types` in `myControlTypes`.
In case you should rename `datetimeControl` in `dateTimeControl`